enclave/core/sgx/init.c

```diff
@@ -41,33 +40,23 @@ static void _check_memory_boundaries(void)
 }
 
 #ifdef OE_WITH_EXPERIMENTAL_EEID
-extern volatile const oe_sgx_enclave_properties_t oe_enclave_properties_sgx;
-extern oe_eeid_t* oe_eeid;
-
-static int _is_eeid_base_image(
-    const volatile oe_sgx_enclave_properties_t* properties)
+static void _check_eeid()
 {
-    return properties->header.size_settings.num_heap_pages == 0 &&
-           properties->header.size_settings.num_stack_pages == 0 &&
-           properties->header.size_settings.num_tcs == 1;
-}
-
-static oe_result_t _eeid_patch_memory()
-{
-    oe_result_t r = OE_OK;
-
-    if (_is_eeid_base_image(&oe_enclave_properties_sgx))
+    if (oe_have_eeid())
     {
-        uint8_t* enclave_base = (uint8_t*)__oe_get_enclave_base();
-        uint8_t* heap_base = (uint8_t*)__oe_get_heap_base();
-        oe_eeid_marker_t* marker = (oe_eeid_marker_t*)heap_base;
-        oe_eeid = (oe_eeid_t*)(enclave_base + marker->offset);
-
-        // Wipe the marker page
-        memset(heap_base, 0, OE_PAGE_SIZE);
+        /* Check that the EEID (meta-)data is within the enclave */
+        const oe_eeid_t* eeid = (oe_eeid_t*)oe_get_eeid();
+        if (!oe_is_within_enclave(eeid, OE_PAGE_SIZE))
+            oe_abort();
+
+        /* Check that the signature is within the EEID page */
+        if (sizeof(oe_eeid_t) + eeid->signature_size > OE_PAGE_SIZE)
+            oe_abort();
+
+        /* Check that the EEID version matches */
+        if (eeid->version != OE_EEID_VERSION)
+            oe_abort();
     }
```

Comment:
Also need to check `eeid->entry_point`

Reply:
Outdated. No measured entry point yet, so nothing to check against.